### PR TITLE
Add openHAB

### PR DESCRIPTION
**Pull request by wborn**
This PR adds support for using the openHAB Docker container in the Rancher community catalog.

The template options used for adding an openHAB stack to Rancher are based on the [openHAB Docker instructions](https://github.com/openhab/openhab-docker).

It will not provide openHAB 2.x as an upgrade option for openHAB 1.8.3 based stacks (due to incompatibilties).

The category of this catalog item is now set to "Automation" which does not yet exist. I.e. similar to that of the pending Home Assistant PR #604. I think it could also be categorized in the "IoT" category.

Comments by @kaikreuzer @cniweb @legacycode are also appreciated. :-)

Below follow some screenshots of the openHAB Rancher catalog entry:

# Searching the catalog

oh-catalog-thumb

# Creating the stack

oh-template

# Showing container logging

oh-logs

### CODE REVIEW

**Comment by cniweb** [comment]
@wborn Great work. Thank you! +1

**Comment by wborn** [comment]
I've also created a [community topic](https://community.openhab.org/t/running-openhab-with-rancher) with details on how to run openHAB with Rancher.

**Comment by kaikreuzer** [comment]
Thanks for picking up the new logo so quickly 👍

**Comment by cniweb** [comment]
Reviewed 5 of 14 files at r1, 1 of 2 files at r2, 8 of 8 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/rancher/community-catalog/621)*

**Comment by cniweb** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/rancher/community-catalog/621#-:-KxAFu7DrySDNYpU4niJ:bnfp4nl)*

File: templates/openhab/0/rancher-compose.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    - variable: "NETWORK_MODE"
+      label: "Network Mode"
+      description: "The network stack to connect the container to"
+      required: true
+      default: "host"
+      type: enum
+      options:
+        - host
+        - bridge
```
Comment (rawmind0) [inline templates/openhab/0/rancher-compose.yml]: What do you think if you also add `managed` network mode option??
```
+        - managed
```

File: templates/openhab/1/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    - variable: "NETWORK_MODE"
+      label: "Network Mode"
+      description: "The network stack to connect the container to"
+      required: true
+      default: "host"
+      type: enum
+      options:
+        - host
+        - bridge
```
Comment (rawmind0) [inline templates/openhab/1/rancher-compose.yml]: What do you think if you also add `managed` network mode option??
```
+        - managed
```

File: templates/openhab/2/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    - variable: "NETWORK_MODE"
+      label: "Network Mode"
+      description: "The network stack to connect the container to"
+      required: true
+      default: "host"
+      type: enum
+      options:
+        - host
+        - bridge
```
Comment (rawmind0) [inline templates/openhab/2/rancher-compose.yml]: What do you think if you also add `managed` network mode option??
```
+        - managed
```

File: templates/openhab/3/rancher-compose.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    - variable: "NETWORK_MODE"
+      label: "Network Mode"
+      description: "The network stack to connect the container to"
+      required: true
+      default: "host"
+      type: enum
+      options:
+        - host
+        - bridge
```
Comment (rawmind0) [inline templates/openhab/3/rancher-compose.yml]: What do you think if you also add `managed` network mode option??
```
+        - managed
```

**Comment by rawmind0** [comment]
Hi @wborn , thanks for the nice PR. :)

Could you take a look to the requested changes?? Please, add `managed` network mode option.

**Comment by wborn** [comment]
Thanks for reviewing this PR @rawmind0! 

Getting the container in managed network mode is done by not setting any `network_mode` at all in docker-compose.yml.tpl right?

**Comment by wborn** [comment]
@rawmind0 I've added the managed network mode and succesfully tested it with Rancher. :slightly_smiling_face:

**Comment by rawmind0** [comment]
@wborn, thank for the changes. :)

LGTM...

**Comment by wborn** [comment]
Thank you too @rawmind0! I'll most likely be putting in a PR soon to also get the next stable openHAB version (2.2.0) added once the images have been uploaded to Docker Hub. :wink:

**Pull request merged**